### PR TITLE
Feature/ use redis secret from subchart

### PR DESCRIPTION
**Pull request by bluebrown**
Signed-off-by: Nico Braun <rainbowstack@gmail.com>

Fixes: #84

### CODE REVIEW

**Review by pierluigilenoci — APPROVED** [review]
LGTM

**Comment by pierluigilenoci** [comment]
@desaintmartin could you please take a look?

File: helm/oauth2-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -112,11 +112,17 @@ spec:
         {{- if eq (default "cookie" .Values.sessionStorage.type) "redis" }}
         - name: OAUTH2_PROXY_SESSION_STORE_TYPE
           value: "redis"
-        {{- if or .Values.sessionStorage.redis.password .Values.sessionStorage.redis.existingSecret }}
+         {{- if or .Values.sessionStorage.redis.existingSecret .Values.sessionStorage.redis.password (and .Values.redis.enabled (.Values.redis.auth).enabled )}}
```
Comment (desaintmartin) [inline helm/oauth2-proxy/templates/deployment.yaml:115]:
```suggestion
        {{- if or .Values.sessionStorage.redis.password .Values.sessionStorage.redis.existingSecret (and .Values.redis.enabled .Values.redis.auth.enabled )}}
```

File: helm/oauth2-proxy/templates/deployment.yaml [hunk 2]
```diff
@@ -112,11 +112,17 @@ spec:
...
         - name: OAUTH2_PROXY_REDIS_PASSWORD
           valueFrom:
             secretKeyRef:
-              name: {{ if .Values.sessionStorage.redis.existingSecret }} {{ .Values.sessionStorage.redis.existingSecret }}{{ else }} {{ template "oauth2-proxy.fullname" . }}-redis-access{{ end }}
+              {{- if .Values.sessionStorage.redis.existingSecret }}
```
Comment (desaintmartin) [inline helm/oauth2-proxy/templates/deployment.yaml:119]: Shouldn't we do the same for sentinel? Note that even without it, this is still an improvement!

**Review by desaintmartin — COMMENTED** [review]
Thanks! See my comments.

File: helm/oauth2-proxy/templates/deployment.yaml (hunk 1 above)
Comment (bluebrown) [reply helm/oauth2-proxy/templates/deployment.yaml:115]: I had it without the extra parenthesis as the beginning, but this is causing issues with Redis is not enabled. Helm errors on the evaluation of `.Values.redis.auth.enabled` for some reason, if the redis chart is not enabled. I would assume it stops early to evaluate when using the, and it's not the case, it seems.

File: helm/oauth2-proxy/templates/deployment.yaml (hunk 2 above)
Comment (bluebrown) [reply helm/oauth2-proxy/templates/deployment.yaml:119]: I guess yes, I didn't understand how this with sentinel works TBH. But I haven't really looked into that. I can take a look later and see if I can figure it out.

**Review by desaintmartin — APPROVED** [review]

**Pull request merged**